ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -150,136 +140,141 @@ def process_result(
 
 
 def leak_message_located(
-    flat_matches_dict: Dict[int, List[ExtendedMatch]],
-    lines: List[Line],
-    padding: int,
-    offset: int,
+    flat_matches: List[Tuple[Line, List[ExtendedMatch]]],
+    is_patch: bool = False,
     clip_long_lines: bool = False,
 ) -> str:
     """
     Display leak message of an incident with location in content.
 
-    :param lines: The lines list
-    :param flat_matches_dict  a dictionary mapping a line number to a list of matches
-    starting at that line.
+    :param flat_matches:  a list of tuples mapping a line to a list of matches starting at that line.
     :param padding: The line padding
     :param offset: The offset due to the line display
+    :param is_patch: Whether the content is a patch
+    :param clip_long_lines: Whether to clip long lines
     """
     leak_msg = StringIO()
+    all_lines = []
+    for line, list_matches in flat_matches:
+        all_lines.append(line)
+        for match in list_matches:
```

Comment:
It's hard to understand why you add lines twice, maybe you can add a comment to explain it :)

Reply:
I went the rout of not adding the line twice as it makes more sense anyway...